api/types/filters/parse.go

```diff
@@ -11,27 +12,34 @@ import (
 	"github.com/docker/docker/api/types/versions"
 )
 
-// Args stores filter arguments as map key:{map key: bool}.
-// It contains an aggregation of the map of arguments (which are in the form
-// of -f 'key=value') based on the key, and stores values for the same key
-// in a map with string keys and boolean values.
-// e.g given -f 'label=label1=1' -f 'label=label2=2' -f 'image.name=ubuntu'
-// the args will be {"image.name":{"ubuntu":true},"label":{"label1=1":true,"label2=2":true}}
+// Args stores a mapping of keys to a set of multiple values.
 type Args struct {
 	fields map[string]map[string]bool
 }
 
-// NewArgs initializes a new Args struct.
-func NewArgs() Args {
-	return Args{fields: map[string]map[string]bool{}}
+// KeyValuePair are used to initialize a new Args
+type KeyValuePair struct {
```

Comment:
Can you explain why we need this?
`NewArgs` can be initialized with a `map[string]string` instead, that leaves a new function `Arg` that  produces one of these, but I'm not sure why someone would use `Arg`. There doesn't seem to be an equivalent before this change.

Reply:
We synced in chat, but answering the question here as well.

A `map[string]string` won't work because `Args` can accept duplicate keys, which causes their values to be added to the same set (`Args.Add()`). So we would have to accept a `map[string]map[string]bool{}` which is a terrible struct to have to initialize. So instead we accept a list of simple key/values pairs.

These pairs should be used to initialize a new `Args` struct, which is common in tests.